src/auth/auth.controller.ts

```diff
@@ -54,4 +61,78 @@ export class AuthController {
 
     res.sendStatus(HttpStatus.OK);
   }
+
+  @ApiExcludeEndpoint()
+  @Get('login')
+  async jwtLogin(
+    @Query(
+      new ValidationPipe({
+        errorHttpStatusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+        transform: true,
+      }),
+    )
+    { token }: LoginQueryDto,
+    @Res() res: Response,
+  ): Promise<void> {
+    if (this.config.get<boolean>('DISABLE_LOGIN')) {
+      throw new UnauthorizedException();
+    }
+
+    const email = this.verifyEmailFromJwt(token);
+
+    const user = await this.usersService.findByEmail(email);
+    if (user) {
+      await this.usersService.updateLastLoginAt(user);
+    } else {
+      throw new UnauthorizedException({ error: 'user_invalid' });
+    }
+
+    this.setResponseHeaders(res, token);
+    res.sendStatus(HttpStatus.OK);
+  }
+
+  private verifyEmailFromJwt(token: string): string {
+    try {
+      const payload = jwt.verify(
+        token,
+        this.config.get<string>('JWT_TOKEN_SECRET'),
+      );
+
+      if (typeof payload === 'string') {
+        throw new UnprocessableEntityException('Invalid payload');
+      }
+
+      if (payload.sub === undefined) {
+        throw new UnprocessableEntityException('Invalid payload object');
+      }
+
+      return payload.sub;
+    } catch (err) {
+      if (err instanceof jwt.JsonWebTokenError) {
+        throw new UnprocessableEntityException({
+          code: err.name,
+          message: err.message,
+        });
+      }
+
+      throw new UnauthorizedException();
+    }
+  }
+
+  private setResponseHeaders(res: Response, token: string): void {
+    const cookieOptions: CookieOptions = {
+      httpOnly: true,
+      maxAge: MS_PER_DAY,
+    };
+
+    if (!this.config.isLocal()) {
+      cookieOptions.domain = this.config.get<string>(
+        'INCENTIVIZED_TESTNET_URL',
+      );
+      cookieOptions.sameSite = 'none';
+      cookieOptions.secure = true;
+    }
+
```

Comment:
Do you want to add the CSRF token here too in the header

Reply:
I was going to do it in a subsequent PR to keep it separate.